src/cluster_legacy.c

```diff
@@ -3121,6 +3122,18 @@ int clusterProcessPacket(clusterLink *link) {
         return 1;
     }
 
+    if (type == CLUSTERMSG_TYPE_MEET && link->node && nodeInHandshake(link->node)) {
+        /* If the link is bound to a node and the node is in the handshake state, and we receive
+         * a MEET packet, it may be that the sender sent multiple MEET packets when reconnecting,
```

Comment:
I don't think this is happening on a reconnect. Since on a reconnect, the link would be new, and therefore `link->node` would be NULL.
From my understanding the crash is happening if the other node sends a MEET packet because it detects that there is no inbound link, this node creates a new node in HANDSHAKE state (with a random node name), and respond with a PONG. The other node receives the PONG and removes the CLUSTER_NODE_MEET flag. This node is supposed to open an outbound connection to the other node in the next cron cycle, but before this happens, the other node re-sends a MEET on the same link because it still detects no inbound connection. And that's when we crash.

Reply:
i guess i want to mention is that it is the other node doing this reconnect, and send two MEET in a short time to myself node.

yes, your understanding is right, my wording is a bit wrong.

Reply:
could you suggest a comment that i can use to put it into the comment? i am a bit suck with the long wording.

Reply:
The new comment looks good. Thank you.